src/catalog_api.cpp

```diff
@@ -75,30 +72,130 @@ static IRCAPIColumnDefinition ParseColumnDefinition(yyjson_val *column_def) {
+	//! Set of recognized config parameters and the duckdb secret option that matches it.
+	static const case_insensitive_map_t<string> config_to_option = {
+		{"s3.access-key-id", "key_id"},
+		{"s3.secret-access-key", "secret"},
+		{"s3.session-token", "session_token"},
+		{"s3.region", "region"},
+		{"s3.endpoint", "endpoint"}
+	};
+
+	auto config_size = yyjson_obj_size(config);
+	if (!config || config_size == 0) {
+		return;
+	}
+	for (auto &it : config_to_option) {
+		auto &key = it.first;
+		auto &option = it.second;
+
+		auto *item = yyjson_obj_get(config, key.c_str());
+		if (item) {
+			options[option] = yyjson_get_str(item);
+		}
+	}
+	auto *access_style = yyjson_obj_get(config, "s3.path-style-access");
+	if (access_style) {
+		string value = yyjson_get_str(access_style);
+		bool path_style;
+		if (value == "true") {
+			path_style = true;
+		} else if (value == "false") {
+			path_style = false;
+		} else {
+			throw InternalException("Unexpected value ('%s') for 's3.path-style-access' in 'config' property", value);
+		}
+		options["use_ssl"] = Value(!path_style);
+		if (path_style) {
+			options["url_style"] = "path";
+		}
+	}
+
+	auto endpoint_it = options.find("endpoint");
+	if (endpoint_it == options.end()) {
+		return;
+	}
+	auto endpoint = endpoint_it->second.ToString();
+	if (StringUtil::StartsWith(endpoint, "http://")) {
+		endpoint = endpoint.substr(7, std::string::npos);
+	}
+	if (StringUtil::EndsWith(endpoint, "/")) {
+		endpoint = endpoint.substr(0, endpoint.size() - 1);
+	}
+	endpoint_it->second = endpoint;
+}
+
 IRCAPITableCredentials IRCAPI::GetTableCredentials(ClientContext &context, IRCatalog &catalog, const string &schema, const string &table, IRCCredentials credentials) {
 	IRCAPITableCredentials result;
 	string api_result = GetTableMetadataCached(context, catalog, schema, table, catalog.secret_name);
 	std::unique_ptr<yyjson_doc, YyjsonDocDeleter> doc(ICUtils::api_result_to_doc(api_result));
 	auto *root = yyjson_doc_get_root(doc.get());
-	auto *warehouse_credentials = yyjson_obj_get(root, "config");
-	auto credential_size = yyjson_obj_size(warehouse_credentials);
 	auto catalog_credentials = IRCatalog::GetSecret(context, catalog.secret_name);
-	if (warehouse_credentials && credential_size > 0) {
-		result.key_id = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.access-key-id", false);
-		result.secret = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.secret-access-key",  false);
-		result.session_token = IcebergUtils::TryGetStrFromObject(warehouse_credentials, "s3.session-token", false);
-		if (catalog_credentials) {
-       		auto kv_secret = dynamic_cast<const KeyValueSecret &>(*catalog_credentials->secret);
-			auto region = kv_secret.TryGetValue("region").ToString();
-			result.region = region;
+
+	// Mapping from config key to a duckdb secret option
+
+	case_insensitive_map_t<Value> config_options;
+	if (catalog_credentials) {
```

Comment:
We probably should only create a secret out of the config if there were no `storage-credentials` provided.